File: GuideObjC/GuideObjC/ViewControllers/RestorePasswordController/RestorePasswordController.m
//
//  RestorePasswordController.m
//  GuideObjC
//
//  Created by LuzanovRoman on 20.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

#import "RestorePasswordController.h"
#import "ProgressController.h"

@interface RestorePasswordController ()
@property (weak, nonatomic) IBOutlet UILabel *errorLabel;
@property (weak, nonatomic) IBOutlet UITextField *emailTextField;
@property (weak, nonatomic) IBOutlet UIButton *restoreButton;
@end

@implementation RestorePasswordController

- (IBAction)restorePassword:(id)sender {
    _errorLabel.text = nil;
    
    if (_emailTextField.text.length == 0) {
        return;
    }
    [self.view endEditing:true];
    
    [ProgressController show];
    [self.graph restorePasswordWithEmail:_emailTextField.text completion:^(id object, NSError * error) {
        [ProgressController hide];
        
        if (error) {
            _errorLabel.text = error.localizedFailureReason;
        }
        else {
            [self performSegueWithIdentifier:@"ResetPassword" sender:nil];
        }
    }];
}

- (IBAction)textDidChange:(UITextField *)sender {
    _restoreButton.enabled = _emailTextField.text.length > 0;
}

// MARK:- UITextFieldDelegate
- (BOOL)textFieldShouldReturn:(UITextField *)textField {
    [textField resignFirstResponder];
    return true;
}
@end

File: GuideSwift/GuideSwift/ViewControllers/RestorePasswordController/RestorePasswordController.swift
//
//  RestorePasswordController.swift
//  GuideSwift
//
//  Created by LuzanovRoman on 20.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

import UIKit

class RestorePasswordController: BaseController, UITextFieldDelegate {
    
    @IBOutlet weak var errorLabel: UILabel!
    @IBOutlet weak var emailTextField: UITextField!
    @IBOutlet weak var restoreButton: UIButton!
    
    @IBAction func restorePassword(_ sender: AnyObject) {
        errorLabel.text = nil
        
        guard let email = emailTextField.text else { return }
        
        view.endEditing(true)
        
        ProgressController.show()
        Graph.restorePassword(withEmail: email) { (object, error) in
            ProgressController.hide()
            
            guard let err = error else {
                self.performSegue(withIdentifier: "ResetPassword", sender: nil)
                return
            }
            self.errorLabel.text = err.localizedFailureReason
        }
    }
    
    @IBAction func textDidChange(_ sender: UITextField) {
        restoreButton.isEnabled = (emailTextField.text ?? "").isEmpty == false
    }
    
    // MARK:- UITextFieldDelegate
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        textField.resignFirstResponder()
        return true
    }
}
